apps/dashboard/app/controllers/batch_connect/session_contexts_controller.rb

```diff
@@ -81,5 +80,22 @@ def session_contexts_param
     # Store session context into a cache file
     def cache_file
       BatchConnect::Session.dataroot(@app.token).tap { |p| p.mkpath unless p.exist? }.join("context.json")
-    end 
+    end
+
+    # Read in context from cache file if cache is disabled and context.json exist
+    def update_session_with_cache
+      cache = cache_file.file? ? JSON.parse(cache_file.read) : {}
+      cache.delete('cluster') if delete_cached_cluster?(cache['cluster'].to_s)
+
+      @session_context.update_with_cache(cache)
+    end
+
+    def delete_cached_cluster?(cached_cluster)
+
+      # if you've cached a cluster that no longer exists
+      !OodAppkit.clusters.include?(cached_cluster) ||
+        # OR the app only has 1 cluster, and it's changed since the previous cluster was cached.
+        # I.e., admin wants to override what you've cached.
+        (@app.clusters.size == 1 && @app.clusters[0] != cached_cluster)
```

Comment:
Why not just do `@app.clusters.include?(cached_cluster)`? Then the above update_session_with_cache would be

```ruby
      cache.delete('cluster') unless @app.clusters.include?(cached_cluster)
```

Reply:
Yep! changed.